src/overlays/effects/ovl_Effect_Ss_Bubble/z_eff_ss_bubble.c

```diff
@@ -5,23 +5,103 @@
  */
 
 #include "z_eff_ss_bubble.h"
+#include "objects/gameplay_keep/gameplay_keep.h"
+
+#define rScale regs[0]
+#define rVecAdjX regs[1]
+#define rVecAdjZ regs[2]
```

Comment:
I'm not following the use of these macros. Are these being used as part of `this->rVecAdjX` for example? How where do regs come from? Are they commonly used? Would a union be better?

Reply:
The general effects struct which all effects use directly, have space for effect specific things that can be used for whatever the effect wants, and can either be set in the init (for example regs[0] here) or used for calculations and other things (regs[1] and regs[2]). The regs array is part of this space so is effect specific, though there is a list of common ones shared between effects in `z64effect.h`.
A union wouldn't scale well since it would have to be a union with every single effect which is just too much.

Reply:
I see, that makes sense then. Just brainstorming another idea, what about something like:
```
typedef enum {
    /* 0 */ R_SCALE,
    /* 1 */ R_VEC_ADJ_X,
    /* 2 */ R_VEC_ADJ_Z
} EffBubbleRegs;
```

And then you could use:
```
this->regs[R_SCALE]
this->regs[R_VEC_ADJ_X]
this->regs[R_VEC_ADJ_Z]
```
So that you don't lose that this is a generic regs struct that's shared between effects. Just brainstorming though, you don't have to adapt this idea

Reply:
Oh I guess to add on to this idea, the enums would need the `EFF_BUBBLE_` prefix, so
```
typedef enum {
    /* 0 */ R_EFF_BUBBLE_SCALE,
    /* 1 */ R_EFF_BUBBLE_VEC_ADJ_X,
    /* 2 */ R_EFF_BUBBLE_VEC_ADJ_Z
} EffBubbleRegs;
```
Or some alternative

Reply:
Interesting idea, I like it. However, this would probably be better in a bigger discord discussion, especially since the way I have it is currently the style of OOT (it's where I got the idea), and I'd prefer consistency.